app/src/main/java/com/example/marcinwisniewski/intivetrainingapp/moviedetails/view/fragment/MovieDetailsFragment.kt

```diff
@@ -8,21 +8,27 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import com.example.marcinwisniewski.intivetrainingapp.R
-import com.example.marcinwisniewski.intivetrainingapp.databinding.DetailsFragmentBinding
-import com.example.marcinwisniewski.intivetrainingapp.moviedetails.viewmodel.DetailsViewModel
+import com.example.marcinwisniewski.intivetrainingapp.databinding.MovieDetailsFragmentBinding
+import com.example.marcinwisniewski.intivetrainingapp.moviedetails.viewmodel.MovieDetailsViewModel
 import com.example.marcinwisniewski.intivetrainingapp.movielist.model.Movie
 
-class DetailsFragment : Fragment() {
-    private lateinit var viewModel: DetailsViewModel
+class MovieDetailsFragment : Fragment() {
+
+    private lateinit var movieDetailsViewModel: MovieDetailsViewModel
+
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
-        val detailsBinding: DetailsFragmentBinding = DataBindingUtil.inflate(layoutInflater, R.layout.details_fragment, container, false)
-        fetchCurrentMovie()
-        detailsBinding.viewModel = viewModel
+        val detailsBinding: MovieDetailsFragmentBinding = DataBindingUtil.inflate(layoutInflater, R.layout.movie_details_fragment, container, false)
+        movieDetailsViewModel = ViewModelProviders.of(this).get(MovieDetailsViewModel::class.java)
+        detailsBinding.movieDetailsViewModel = movieDetailsViewModel
         return detailsBinding.root
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        fetchCurrentMovie()
+    }
+
     private fun fetchCurrentMovie() {
-        viewModel = ViewModelProviders.of(this).get(DetailsViewModel::class.java)
-        viewModel.fetchChoosenMovie(arguments?.getSerializable(getString(R.string.current_movie)) as Movie?)
+        movieDetailsViewModel.fetchChoosenMovie(arguments?.getSerializable(getString(R.string.current_movie)) as Movie?)
```

Comment:
Add TODO how to avoid optional Movie?. Should be Movie

Reply:
Fixed in upcoming commit